posthog/src/main/java/com/posthog/PostHog.kt

```diff
@@ -415,7 +434,11 @@ public class PostHog private constructor(
 
         if (previousDistinctId != distinctId) {
             // We keep the AnonymousId to be used by decide calls and identify to link the previousId
-            this.anonymousId = previousDistinctId
+            if (previousDistinctId.isNotBlank()) {
```

Comment:
bit confused, do we need these checks everywhere? Can we prevent this at a higher level, like not setting `this.distinctId` to be an invalid id ever?

Having this check here seems sus to me, since it's not really actionable by the user anymore

Reply:
I see gotcha!